pkg/cloud-init/cloud-init.go

```diff
@@ -206,7 +206,7 @@ func GenerateLocalData(vmName string, namespace string, source *v1.CloudInitNoCl
 	} else {
 		return errors.New(fmt.Sprintf("userDataBase64 or userData is required for no-cloud data source"))
 	}
-	metaData := []byte(fmt.Sprintf("{ \"instance-id\": \"%s.%s\" }\n", vmName, namespace))
+	metaData := []byte(fmt.Sprintf("instance-id: %s.%s\nlocal-hostname: %s.%s\n", vmName, namespace, vmName, namespace))
```

Comment:
I originally used yaml for metadata and I don't believe cirros accepted it. That's why json was used.

Now that the networking is complete, we may want to consider using whatever host name is associated with the pod as the local hostname. Ideally the hostname a VM see's internally for itself is the same one other vms could use to contact the VM (if dns is in use of course)

Reply:
I think we should create our own dns subdomain. See: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods-hostname-and-subdomain-fields

We would set the pods `hostName` to the name of the vm and set a `subdomain` on the pod which we could simply call `virtualmachines` or `vms` or `vm`. Then we can simply set the hostname of the vm to the name of the vm. Further we will have reasonable dns entries which will point to our vms. For instance a FQDN like `myvm.virtualmachines.my-namespace.svc.cluster.local`

Reply:
Here a POC: #784. @davidvossel so this basically makes sure that we can find the pod with the vm name, which could be more valuable than the other way round (setting the pod name inside the vm and having to reach the vm via the pod-dns entry).

Reply:
@rmohr i think you're right. The VM name makes more sense than the Pod name. I didn't actually know that we had control over that. Great catch!

So, for this PR, we just need to make sure we converge on how we construct the host name.